config/config_cmd.go

```diff
@@ -75,6 +99,155 @@ func redirectStdOut() (old, new *os.File) {
 	return
 }
 
+func printConfigTable(section Section, sectionName string, isSubsection bool, visitedSection map[string]bool, typeLink map[reflect.Type]string) {
+	val := reflect.Indirect(reflect.ValueOf(section.GetConfig()))
+
+	if val.Kind() == reflect.Slice {
+		val = reflect.Indirect(reflect.ValueOf(val.Index(0).Interface()))
+	}
+
+	table := tablewriter.NewWriter(os.Stdout)
+	table.SetHeader([]string{"Name", "Type", "Default Value", "Description"})
+	table.SetAlignment(4)
+	table.SetRowLine(true)
+
+	subsections := make(map[string]interface{})
+	for i := 0; i < val.Type().NumField(); i++ {
+		t := val.Type().Field(i)
+		tagType := t.Type
+		if tagType.Kind() == reflect.Ptr {
+			tagType = t.Type.Elem()
+		}
+
+		fieldName := t.Name
+		fieldType := tagType.Kind().String()
+		fieldDefaultValue := fmt.Sprintf("%v", val.Field(i))
+		fieldDescription := ""
+
+		if tagType.Kind() == reflect.Map || tagType.Kind() == reflect.Slice || tagType.Kind() == reflect.Struct {
+			fieldType = tagType.String()
+			// Set default value to field type, and user can check its default value in subsection table
+			fieldDefaultValue = fieldType
+		}
+
+		if jsonTag := t.Tag.Get("json"); len(jsonTag) > 0 && !strings.HasPrefix(jsonTag, "-") {
+			var commaIdx int
+			if commaIdx = strings.Index(jsonTag, ","); commaIdx < 0 {
+				commaIdx = len(jsonTag)
+			}
+			if jsonTag[:commaIdx] != "" {
+				fieldName = jsonTag[:commaIdx]
+			}
+		}
+
+		if pFlag := t.Tag.Get("pflag"); len(pFlag) > 0 && !strings.HasPrefix(pFlag, "-") {
```

Comment:
I think you can use [this class](https://github.com/flyteorg/flytestdlib/blob/master/cli/pflags/api/tag.go) to parse the tag for you... and extract the information you need... feel free to add the trimPrefix call there too..

Reply:
Can't import it. will get an error. `import cycle not allowed`